### ISSUE TITLE
Selected table can change during delete process

### ISSUE THREAD

**Issue by MoonWatcher582** [sequelpro/sequelpro#2742]
After selecting the `Delete` option on the delete table prompt, pressing a (letter) key will cause the application to select the first table that begins with that letter. This deletes the newly selected table, rather than the original table that ought to be deleted.

Steps to Reproduce:

1. Create two tables `test` and `zed`
2. Right click on `test` and select `Delete Table`
3. Confirm by clicking `Delete` and immediately press the `Z` key

Expected Results:

1. `test` should be deleted

Actual Results:

1. `zed` should now be deleted rather than `test`

**Comment by dmoagx** [comment]
Can reproduce in 1.1.2 and current nightly.

**Comment by dmoagx** [comment]
This happens because we use a timer callback instead of calling the method directly in the `sheetDidEnd:…` callback, which gives the key event the chance to slip in between:
```objc
			[self performSelector:@selector(_removeTable:) 
					   withObject:[NSNumber numberWithInteger:[[(NSAlert *)sheet suppressionButton] state]] 
					   afterDelay:0.0];
```
This doesn't really make sense to me.

**Comment by stuconnolly** [comment]
Yeah, seems a bit strange seeing as the delay is set to zero. Was added in 5a2bb94 so maybe it was to aid debugging and was never supposed to be committed.

I say remove it.

**Comment by bmoyles0117** [comment]
Is it possible that the delay was introduced to prevent the request from blocking the main UI thread, in the event of some network connectivity failure? What would happen if you deleted a table, and the connection was not established at that time, would the deletion prevent the "Reconnect" UI element to pop up?

**Comment by dmoagx** [comment]
@bmoyles0117 I don't think so, since the delete operation is still executed on the main thread anyway.
